tests/Integration/ApiPlatform/ModuleEndpointTest.php

```diff
@@ -646,6 +651,90 @@ public function testBulkUninstallModule()
         }
     }
 
+    /**
+     * @depends testInstallModule
+     */
+    public function testUpgradeModule(): void
+    {
+
+        $module212 = array(
+            'moduleId' => null,
+            'technicalName' => 'dashproducts',
+            'source' => 'https://github.com/PrestaShop/dashproducts/releases/download/v2.1.2/dashproducts.zip',
+            'version' => '2.1.2',
+            // Module is simply uploaded not installed
+            'installedVersion' => null,
+            'enabled' => false,
+            'installed' => false,
+
+        );
+        $bearerToken = $this->getBearerToken(['module_write']);
+
+        // Upload Zip from github
+        static::createClient()->request('POST', '/module/upload-source', [
+            'auth_bearer' => $bearerToken,
+            'json' => [
+                'source' => $module212['source'],
+            ],
+        ]);
+
+        static::createClient()->request('PUT', sprintf('/module/%s/install', $module212['technicalName']), [
+            'auth_bearer' => $bearerToken,
+            // We must define a JSON body even if it is empty, we need to search how to make this optional
+            'json' => [
+            ],
+        ]);
+
+        $module213 = array(
+            'moduleId' => null,
+            'technicalName' => 'dashproducts',
+            'source' => 'https://github.com/PrestaShop/dashproducts/releases/download/v2.1.3/dashproducts.zip',
+            'version' => '2.1.3',
+            // Module is simply uploaded not installed
+            'installedVersion' => '2.1.2',
+            'enabled' => false,
+            'installed' => false,
+
+        );
+
+        static::createClient()->request('POST', '/module/upload-source', [
+            'auth_bearer' => $bearerToken,
+            'json' => [
+                'source' => $module213['source'],
+            ],
+        ]);
+
+        $response = static::createClient()->request('POST', sprintf('/module/%s/upgrade', $module212['technicalName']), [
```

Comment:
```suggestion
        $response = static::createClient()->request('PUT', sprintf('/module/%s/upgrade', $module212['technicalName']), [
```
That's why the integration tests are failing, the endpoint is only accessible via the `PUT` method